File: problems/002-two-sum/java/Practice.java
import java.util.*;
public class Practice implements Solution {
    public Object solve(Map<String, Object> c) {
        List<Object> nums = (List<Object>) c.get("nums");
        long target = ((Number) c.get("target")).longValue();
        Map<Long, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.size(); i++) {
            long n = ((Number) nums.get(i)).longValue();
            if (seen.containsKey(target - n)) return List.of(seen.get(target - n), i);
            seen.put(n, i);
        }
        return List.of();
    }
}

File: problems/002-two-sum/typescript/practice.ts
export function solve(c: any): number[] | null {
  const seen: Record<number, number> = {};
  for (let i = 0; i < c.nums.length; i++) {
    const need = c.target - c.nums[i];
    if (need in seen) return [seen[need], i];
    seen[c.nums[i]] = i;
  }
  return null;
}
